### PR TITLE
Basic ha e2e test for core contrail services

### PR DESCRIPTION
**Pull request by shifali19**

### CODE REVIEW

File: test/e2e/ha/scale_core_contrail_services_test.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+		t.Run("when replicas is set to 3 in manager", func(t *testing.T) {
+			var replicas int32 = 3
+			_, err := controllerutil.CreateOrUpdate(context.Background(), f.Client.Client, cluster, func() error {
+				cluster.Spec.CommonConfiguration.Replicas = &replicas
+				return nil
+			})
+
+			assert.NoError(t, err)
+
+			w := wait.Wait{
```
Comment (katrybacka) [inline test/e2e/ha/scale_core_contrail_services_test.go:273]: `w := wait.Wait` can be extracted from these t.Runs and defined once

File: test/e2e/ha/scale_core_contrail_services_test.go [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+				Namespace:     namespace,
+				Timeout:       waitTimeout,
+				RetryInterval: retryInterval,
+				KubeClient:    f.KubeClient,
+				Logger:        log,
+			}
+
+			t.Run("then all services are scaled up from 1 to 3 node", func(t *testing.T) {
+				t.Run("then a ready Zookeeper StatefulSet should be created", func(t *testing.T) {
```
Comment (katrybacka) [inline test/e2e/ha/scale_core_contrail_services_test.go:282]: the name of these assertions should be more like then Zookeeper STS is scaled and ready, same below

File: test/e2e/ha/scale_core_contrail_services_test.go [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+		})
+
+		t.Run("when replicas is set to 3 in manager", func(t *testing.T) {
+			var replicas int32 = 3
+			_, err := controllerutil.CreateOrUpdate(context.Background(), f.Client.Client, cluster, func() error {
+				cluster.Spec.CommonConfiguration.Replicas = &replicas
+				return nil
+			})
+
+			assert.NoError(t, err)
```
Comment (katrybacka) [inline test/e2e/ha/scale_core_contrail_services_test.go:271]: I think that we should require NoError here if the update was a failure there is no point in further checking

File: test/e2e/ha/scale_core_contrail_services_test.go (hunk 1 above)
Comment (shifali19) [reply test/e2e/ha/scale_core_contrail_services_test.go]: Done

File: test/e2e/ha/scale_core_contrail_services_test.go (hunk 2 above)
Comment (shifali19) [reply test/e2e/ha/scale_core_contrail_services_test.go]: Done

File: test/e2e/ha/scale_core_contrail_services_test.go (hunk 3 above)
Comment (shifali19) [reply test/e2e/ha/scale_core_contrail_services_test.go]: Done

File: test/e2e/ha/scale_core_contrail_services_test.go [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+
+				t.Run("then Cassandra StatefulSet should be scaled and ready", func(t *testing.T) {
+					assert.NoError(t, w.ForReadyStatefulSet("hatest-cassandra-cassandra-statefulset", replicas))
+				})
+
+				t.Run("then Config StatefulSet should be scaled and ready", func(t *testing.T) {
+					assert.NoError(t, w.ForReadyStatefulSet("hatest-config-config-statefulset", replicas))
+				})
+
+				t.Run("then webui StatefulSet should be scaled and ready", func(t *testing.T) {
```
Comment (katrybacka) [inline test/e2e/ha/scale_core_contrail_services_test.go:286]: WebUI instead of webui

Comment (shifali19) [reply test/e2e/ha/scale_core_contrail_services_test.go]: Done

**Pull request merged**